client/network/sync/src/engine.rs

```diff
@@ -590,6 +608,33 @@ where
 			self.tick_timeout.reset(TICK_TIMEOUT);
 		}
 
+		// go over all connected peers and check if any of them have been idle for a while. Idle in
+		// this case means that we haven't sent or received block announcements to/from this peer.
+		// If that is the case, because of #5685, it could be that the block announces substream is
+		// not actually open and and this peer is just wasting a slot and is should be replaced with
+		// some other node that is willing to send us block announcements.
+		for (id, peer) in self.peers.iter() {
```

Comment:
Last nitpick would be to put this as part of the `tick_timeout` above. So, this loop is not running every time we call `poll`.